# Copyright 2018 Stanford University
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

diff --git a/Jamroot b/Jamroot
index 94494bf..e06d6df 100755
--- a/Jamroot
+++ b/Jamroot
@@ -27,50 +27,50 @@ if [ os.name ] = NT
         errors.user-error "*** BOOST_ROOT must be set to a Boost installation." ;
     }
 
-    LUA_PATH = [ os.environ LUA_PATH ] ;
+    LUAJIT_PATH = [ os.environ LUAJIT_PATH ] ;
 
-    if ! $(LUA_PATH)
+    if ! $(LUAJIT_PATH)
     {
-        errors.user-error "*** LUA_PATH must be set." ;
+        errors.user-error "*** LUAJIT_PATH must be set." ;
     }
 
-    local includes = [ GLOB $(LUA_PATH)/include $(LUA_PATH)/src : lua.h ] ;
+    local includes = [ GLOB $(LUAJIT_PATH)/include $(LUAJIT_PATH)/src : lua.h ] ;
 
     if ! $(includes)
     {
-        errors.user-error "*** Unable to determine Lua include directory." ;
+        errors.user-error "*** Unable to determine LuaJIT include directory." ;
     }
 
     includes = $(includes:D) ;
 
     debug-message found include directory in $(includes) ;
 
-    if [ GLOB $(LUA_PATH) : lib ]
+    if [ GLOB $(LUAJIT_PATH) : lib ]
     {
-        lib lua : : <link>static <name>lua5.1 <search>$(LUA_PATH)/lib/static : : <include>$(includes) ;
-        lib lua : : <link>shared <name>lua5.1 <search>$(LUA_PATH)/lib : : <include>$(includes) ;
+        lib lua : : <link>static <name>luajit-5.1 <search>$(LUAJIT_PATH)/lib : : <include>$(includes) ;
+        lib lua : : <link>shared <name>luajit-5.1 <search>$(LUAJIT_PATH)/lib : : <include>$(includes) ;
     }
-    else if [ GLOB $(LUA_PATH) : lua5.1.dll ]
+    else if [ GLOB $(LUAJIT_PATH) : luajit-5.1.dll ]
     {
-        lib lua : : <name>lua5.1 <search>$(LUA_PATH) : : <include>$(includes) ;
+        lib lua : : <name>luajit-5.1 <search>$(LUAJIT_PATH) : : <include>$(includes) ;
     }
 }
 else if [ os.name ] in LINUX MACOSX FREEBSD
 {
-    LUA_PATH = [ os.environ LUA_PATH ] ;
+    LUAJIT_PATH = [ os.environ LUAJIT_PATH ] ;
     HOME = [ os.environ HOME ] ;
 
     local possible-prefixes =
-        $(LUA_PATH) $(HOME)/Library/Frameworks /Library/Frameworks /usr /usr/local /opt/local /opt ;
+        $(LUAJIT_PATH) $(HOME)/Library/Frameworks /Library/Frameworks /usr /usr/local /opt/local /opt ;
 
     local possible-suffixes =
-        include/lua5.1 include/lua51 include/lua include ;
+        include/luajit-2.0 include/luajit include ;
 
     local includes = [ GLOB $(possible-prefixes)/$(possible-suffixes) : lua.h ] ;
 
     if ! $(includes)
     {
-        errors.user-error "*** Unable to determine Lua include directory." ;
+        errors.user-error "*** Unable to determine LuaJIT include directory." ;
     }
 
     local includes = $(includes[1]:D) ;
@@ -83,8 +83,8 @@ else if [ os.name ] in LINUX MACOSX FREEBSD
 
     local lib = $(prefix)/lib ;
 
-    local names = liblua5.1 liblua51 liblua ;
-    local extensions = .a .so ;
+    local names = libluajit-5.1 libluajit51 libluajit ;
+    local extensions = .a .so .dylib ;
 
     library = [ GLOB $(lib)/lua51 $(lib)/lua5.1 $(lib)/lua $(lib) :
         $(names)$(extensions) ] ;
@@ -98,7 +98,7 @@ else if [ os.name ] in LINUX MACOSX FREEBSD
 
     if ! $(lib-name)
     {
-        errors.user-error "*** Unable to find Lua library." ;
+        errors.user-error "*** Unable to find LuaJIT library." ;
     }
 
     lib m : : <link>shared ;
